### PR TITLE
fix(argo-cd): deprecate server.extraArgs."--insecure"

### PR DESCRIPTION
**Pull request by aslafy-z**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -27,3 +27,4 @@ annotations:
     - "[Deprecated]: Configuration option configs.gpgKeys"
     - "[Deprecated]: Configuration option configs.gpgKeysAnnotations"
     - "[Fixed]: Documentation for declarative setup"
```
Comment (mbevc1) [inline charts/argo-cd/Chart.yaml:29]: Could you please remove old changes as it gets merged when published?

Comment (aslafy-z) [reply charts/argo-cd/Chart.yaml]: Done

**Comment by pdrastil** [comment]
FYI - Technically this is not deprecated as command line args have always higher priority than argocd-params-cmd configmap, however I get the intent :)

**Comment by aslafy-z** [comment]
@pdrastil Yeah, I was not sure about the wording. Do you want me to change the message?

**Comment by pdrastil** [comment]
@aslafy-z I'm fine with wording as it directs users to adopt the `params` section. This is just info that `args` should not be removed as they are complementary feature.

**Review by mbevc1 — APPROVED** [review]

**Comment by mbevc1** [comment]
Thanks for the contribution

**Pull request merged**